test/e2e/tests/test_trail.py

```diff
@@ -0,0 +1,84 @@
+# Copyright Amazon.com Inc. or its affiliates. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License"). You may
+# not use this file except in compliance with the License. A copy of the
+# License is located at
+#
+#	 http://aws.amazon.com/apache2.0/
+#
+# or in the "license" file accompanying this file. This file is distributed
+# on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+# express or implied. See the License for the specific language governing
+# permissions and limitations under the License.
+
+"""Integration tests for the CloudTrail Trail resource
+"""
+
+import boto3
+import logging
+import time
+from typing import Dict
+
+import pytest
+
+from acktest.k8s import resource as k8s
+from acktest.resources import random_suffix_name
+from e2e import service_marker, CRD_GROUP, CRD_VERSION, load_cloudtrail_resource
+from e2e.replacement_values import REPLACEMENT_VALUES
+from e2e.bootstrap_resources import get_bootstrap_resources
+from e2e.tests.helper import CloudTrailValidator
+from e2e.common import TRAIL_NAME
+
+RESOURCE_PLURAL = "trails"
+
+CREATE_WAIT_AFTER_SECONDS = 10
+MODIFY_WAIT_AFTER_SECONDS = 10
+DELETE_WAIT_AFTER_SECONDS = 10
+
+@service_marker
+@pytest.mark.canary
+class TestTrail:
+    def test_create_delete(self, cloudtrail_client):
```

Comment:
Creating the CR, asserting and then deleting all within a single test can leave resources hanging. Please see the [EKS Cluster tests](https://github.com/aws-controllers-k8s/eks-controller/blob/main/test/e2e/tests/test_cluster.py#L57-L88) for how I use fixtures to get around this. In the current format, if the assertions fail, the delete code will never be called and we will have an account full of Trail resources.

Reply:
Cool! Adding this